Update assumption on lang codes to explicitly reject ISO 639.2

Jym77 opened the pull request:
Update assumption on `lang` attribute and grandfathered tags to all non-valid tags, and explicitly mention that ISO 639.2 three-letters codes are bad.

Closes issue(s):

- closes #1613 

Need for Final Call:
This will not require a Final Call (changing an assumption)

---

## Pull Request Etiquette

### **When creating PR:**

- [X] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).
- [X] Make sure you do not remove the "How to Review and Approve" section in your pull request description

### **After creating PR:**

- [X] Add yourself (and co-authors) as "Assignees" for PR.
- [X] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [X] [Link the PR to any issue it solves](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). This will be done automatically by [referencing the issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) at the top of this comment in the indicated place.
- [X] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

### **When merging a PR:**

- [ ] Close any issue that the PR resolves. This will happen automatically upon merging if the PR was correctly linked to the issue, e.g. by referencing the issue at the top of this comment.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Final Call period. In case of disagreement, the longer period wins.

WilcoFiers commented on _rules/element-lang-matches-default-language-off6ek.md:
```diff
@@ -46,7 +46,7 @@ For each test target, the [primary language][] of its `lang` [attribute value][]
 - This rule assumes that user agents and assistive technologies can programmatically determine [valid language tags][] even if these do not conform to the [BCP 47][] syntax.
 
-- This rule assumes that [grandfathered tags][] are not used as these will not be recognized as [valid language tags][].
+- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.2 Language of Parts][sc312]; this notably excludes [grandfathered tags][] or [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```
```suggestion
- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.2 Language of Parts][sc312]; this notably excludes [grandfathered tags][] and [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```

WilcoFiers commented on _rules/element-lang-valid-de46e4.md:
```diff
@@ -43,7 +43,7 @@ For each test target, the `lang` [attribute value][] is a [valid language tag][]
 - This rule assumes that user agents and assistive technologies can programmatically determine [valid language tags](#valid-language-tag) even if these do not conform to the [BCP 47][] syntax.
 
-- This rule assumes that [grandfathered tags][] are not used as these will not be recognized as [valid language tags](#valid-language-tag).
+- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.2 Language of Parts][sc312]; this notably excludes [grandfathered tags][] or [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```
```suggestion
- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.2 Language of Parts][sc312]; this notably excludes [grandfathered tags][] and [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```

WilcoFiers commented on _rules/html-page-lang-matches-default-ucwvc8.md:
```diff
@@ -50,7 +50,7 @@ For each test target, the [primary language][] of the [valid language tag][] mat
 - This rule assumes that user agents and assistive technologies can programmatically determine [valid language tags](#valid-language-tag) even if these do not conform to the [BCP 47][] syntax.
 
-- This rule assumes that [grandfathered tags][] are not used as these will not be recognized as [valid language tags](#valid-language-tag).
+- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.1 Language of Page][sc311]; this notably excludes [grandfathered tags][] or [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```
```suggestion
- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.1 Language of Page][sc311]; this notably excludes [grandfathered tags][] and [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```

WilcoFiers commented on _rules/html-page-lang-valid-bf051a.md:
```diff
@@ -42,7 +42,7 @@ For each test target, the `lang` attribute has a [valid language tag][].
 - This rule assumes that user agents and assistive technologies can programmatically determine [valid language tags](#valid-language-tag) even if these do not conform to the [BCP 47][] syntax.
 
-- This rule assumes that [grandfathered tags][] are not used as these will not be recognized as [valid language tags](#valid-language-tag).
+- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.1 Language of Page][sc311]; this notably excludes [grandfathered tags][] or [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```
```suggestion
- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.1 Language of Page][sc311]; this notably excludes [grandfathered tags][] and [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```

WilcoFiers commented on _rules/html-page-lang-xml-lang-match-5b7ae0.md:
```diff
@@ -38,7 +38,7 @@ For each test target, the values of the [primary language subtags][], if any exi
 - This rule assumes that user agents and assistive technologies can programmatically determine [valid language tags](#valid-language-tag) even if these do not conform to the [BCP 47][] syntax.
 
-- This rule assumes that [grandfathered tags][] are not used as these will not be recognized as [valid language tags](#valid-language-tag).
+- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.1 Language of Page][sc311]; this notably excludes [grandfathered tags][] or [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```
```suggestion
- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.1 Language of Page][sc311]; this notably excludes [grandfathered tags][] and [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```

WilcoFiers reviewed (dismissed):
Kinda think we should add some failure examples to specifically check that lang="eng" is failed. Probably same for grandfather tags too. WDYT?

Jym77 commented:
> Kinda think we should add some failure examples to specifically check that lang="eng" is failed. Probably same for grandfather tags too. WDYT?

@WilcoFiers @ajanec01 @carlosapaduarte I've added examples with grandfathered and 3-letters codes.

I also noticed that all the links to BCP went to a generic BCP 47 page. I've updated them to RFC 5646. This is not perfect (as BCP numbers are supposed to be more stable than RFC ones), but should be good enough (this one has been here since 2009). Couldn't find a page with the full BCP 47 **which would let me deep link to specific sections** 😕 

---
Adding these examples means the following for tools who actually accept them:
- for the "`lang` is valid" rules, there are some Failed Examples that they may pass. Thus they won't have a correct implementation.
- for the "`lang` match default" rules, there are some Inapplicable Examples that they may pass. This won't really affect the implementation.

WilcoFiers commented on _rules/element-lang-matches-default-language-off6ek.md:
```diff
@@ -299,17 +299,43 @@ There is no [text inheriting its programmatic language][] from this `p` element
 
+#### Inapplicable Example 6
+
+The `lang` [attribute value][] of this `p` element is an [iso 639.2][] three letters code, hence not a [valid language tag][].
```
```suggestion
The `lang` [attribute value][] of this `p` element is an [iso 639.2][] three letters code. This is not a [valid language tag][].
```

WilcoFiers commented on _rules/element-lang-matches-default-language-off6ek.md:
```diff
@@ -299,17 +299,43 @@ There is no [text inheriting its programmatic language][] from this `p` element
+
+#### Inapplicable Example 7
+
+The `lang` [attribute value][] of this `p` element is a [grandfathered tag][grandfathered tags], hence not a [valid language tag][].
```
```suggestion
The `lang` [attribute value][] of this `p` element is a [grandfathered tag][grandfathered tags]. This is not a [valid language tag][].
```

WilcoFiers commented on _rules/element-lang-matches-default-language-off6ek.md:
````diff
@@ -299,17 +299,43 @@ There is no [text inheriting its programmatic language][] from this `p` element
+```html
+<html lang="lb">
+	<body>
+		<p lang="i-lux">Lëtzebuerg ass e Land an Europa.</p>
````
Owww, good one! I like that better than having Klingon in our examples!

WilcoFiers commented on _rules/element-lang-valid-de46e4.md:
```diff
@@ -238,6 +238,30 @@ This `div` element has an invalid `lang` [attribute value][]. The [accessible na
 
+#### Failed Example 8
+
+The `lang` [attribute value][] of this `p` element is an [iso 639.2][] three letters code, hence not a [valid language tag][].
```
How about making this a little more explicit:

```suggestion
The `lang` [attribute value][] of this `p` element is an [iso 639.2][] three letters code. This is not part of the [language subtag registry][].
```

Jym77 replied:
I think we should stick with the "not a valid tag" explanation as we do for grandfathered tags (https://github.com/act-rules/act-rules.github.io/pull/1650/files#r673783251) (for the sake of consistency). WDYT?
```suggestion
The `lang` [attribute value][] of this `p` element is an [iso 639.2][] three letters code. This is not a [valid language tag][].
```

WilcoFiers commented on _rules/element-lang-valid-de46e4.md:
```diff
@@ -238,6 +238,30 @@ This `div` element has an invalid `lang` [attribute value][]. The [accessible na
+
+#### Failed Example 9
+
+The `lang` [attribute value][] of this `p` element is a [grandfathered tag][grandfathered tags], hence not a [valid language tag][].
```
```suggestion
The `lang` [attribute value][] of this `p` element is a [grandfathered tag][grandfathered tags]. This is not part of the [language subtag registry][].
```

WilcoFiers commented on _rules/html-page-lang-valid-bf051a.md:
````diff
@@ -97,6 +97,30 @@ This `html` element has a `lang` attribute whose value is not a [valid language
+```html
+<html lang="eng">
+	<body>
+		<p lang="en">I love ACT rules!</p>
````
What's the reason for putting in the `lang` on this `p` element? I think this is clearer as an example if it isn't there. One could even make the case that in this particular example it isn't actually an error. Not that I want to have that discussion.

```suggestion
		<p>I love ACT rules!</p>
```

Jym77 replied:
Without the `lang` attribute on the paragraph, the page is also failing 3.1.2 (on top of 3.1.1 that the rule test).

WilcoFiers commented on _rules/html-page-lang-valid-bf051a.md:
````diff
@@ -97,6 +97,30 @@ This `html` element has a `lang` attribute whose value is not a [valid language
+```html
+<html lang="i-lux">
+	<body>
+		<p lang="lb">Lëtzebuerg ass e Land an Europa.</p>
````
```suggestion
		<p>Lëtzebuerg ass e Land an Europa.</p>
```

WilcoFiers reviewed (changes requested):
Great work! Just a couple of editorial suggestions.
